### PR TITLE
meta-smartphone: Allow building of hammerhead (mainline) and hammerhead (halium)

### PR DESCRIPTION
**Pull request by Herrie82**
Use "hammerhead-halium" to build the Halium/Android based image and "hammerhead" for the mainline kernel version.

Useful for building and comparing side by side while migrating to mainline kernel. 

Signed-off-by: Herman van Hazendonk <github.com@herrie.org>

### CODE REVIEW

**Comment by Herrie82** [comment]
@shr-project @Tofee Build tested for both machines

File: meta-android/recipes-core/images/initramfs-android-image.bb [hunk 1]
```diff
@@ -8,9 +8,9 @@ ANDROID_EXTRA_INITRAMFS_IMAGE_INSTALL ?= ""
 
 VIRTUAL-RUNTIME_android-initramfs-scripts ?= "initramfs-scripts-android"
 
-IMAGE_INSTALL = "busybox base-passwd android-tools bash"
-IMAGE_INSTALL += "${VIRTUAL-RUNTIME_android-initramfs-scripts}"
-IMAGE_INSTALL += "${ANDROID_EXTRA_INITRAMFS_IMAGE_INSTALL}"
+IMAGE_INSTALL = "busybox busybox-mdev base-passwd android-tools bash"
+IMAGE_INSTALL += " ${VIRTUAL-RUNTIME_android-initramfs-scripts}"
```
Comment (shr-project) [inline meta-android/recipes-core/images/initramfs-android-image.bb:12]: The leading space doesn't need to be added.

Comment (Herrie82) [reply meta-android/recipes-core/images/initramfs-android-image.bb]: @shr-project https://github.com/shr-distribution/meta-smartphone/pull/144/commits/0cc1008eaba923964d6dd401abf04ab7be9cc466

**Comment by shr-project** [comment]
Merged in both kirkstone and master with small modifactions, except the 5.19-rc7 commit which I've updated to use shr-distribution/linux repo as well.

**Pull request merged**